[wip] use github auth

jacobian opened the pull request:
Rather than using Django's built-in auth, admins should auth against github.
- [x] add github auth
- [x] use `GithubTeamOAuth2` instead of `GithubOrganizationOAuth2`
- [ ] create the signup flow:
  - [ ] start page for creating an account
  - [ ] post-auth complete page
  - [ ] document how admins should add more perms to created users
- [ ] add "login with github" to admin page
- [ ] remove password auth entirely
- [ ] figure out and document initial setup flow
- [ ] add whitelist?
- [ ] test with other users to make sure this is right

jacobian commented:
Or maybe we should use github instead? /cc @adelevie

NoahKunin commented:
Why is MyUSA being considered? It's on the line to deprecation due to Login. I'd use GitHub auth until Login is up.

jacobian commented:
github it is 👍

Commit:
installed python-social-auth

Commit:
Get social settings from env

Commit:
I accidentally a punctuation

Commit:
Merge branch 'master' into gh-auth

Commit:
Merge branch 'master' into gh-auth

Commit:
fix flake8 error

jessieay commented:
@jacobian thoughts on scoping to https://github.com/orgs/18F/teams/18f so only 18F staff can auth rather than anyone in the 18F org? (197 vs 346 ppl)

jacobian commented:
@jessieay it's hard to tell, but that's what this does, see https://github.com/18F/bug-bounty/pull/6/files#diff-041953c637f95e25fb35d51022891763R79 - using the `GithubOrganizationOAuth2` backend takes care of that. See http://psa.matiasaguirre.net/docs/backends/github.html#github-for-organizations for the (scant) documentation.

jessieay commented:
@jacobian I was thinking you'd want `GithubTeamOAuth2.` -- I think you are using the 18F Org, which is bigger than the 18F Team

jacobian commented:
@jessieay oh shit, I'd missed that difference! Yes, you're totally right - thank you!

konklone commented:
@jacobian So how much do we pay @jessieay for that one? :smile_cat:

NoahKunin commented:
@jessieay http://i.giphy.com/13B1WmJg7HwjGU.gif

Commit:
Use GithubTeamOAuth2 instead of GithubOrganizationOAuth2 (thanks @jessieay)

Note that this means the oauth app now needs read:org permissions, which
means the app'll need to whitelisted when we're ready to deploy this.

Commit:
fixed flake8 errors

jessieay commented on bugbounty/settings.py:
```diff
@@ -70,3 +73,21 @@
+SOCIAL_AUTH_GITHUB_TEAM_KEY = os.environ.get('GITHUB_KEY', None)
+SOCIAL_AUTH_GITHUB_TEAM_SECRET = os.environ.get('GITHUB_SECRET', None)
+
+# The team ID doesn't seem to be easy to find on the web anywhere, so I
```
agreed that it's hard to find. I added some docs in dolores on how I did it: https://github.com/18F/dolores-landingham-bot/compare/jy-github-auth#diff-6a3371457528722a734f3c51d9238c13R127

Not sure if that seems easier or harder than what you're recommending
